app.py

```diff
@@ -2,7 +2,86 @@
 
 app = create_app()
 
-@app.route('/')
-def hello():
-  return "Hello, world!"
+
+# @app.route("/")
+# def hello():
+#   return "Hello, world"
+
+# API access variables and fetch stuff -- will move to another file later
+import os
+import requests
+import flask
+from dotenv import load_dotenv
+from authlib.integrations.flask_client import OAuth
+
+load_dotenv()
+
+oauth = OAuth(app)
+# Note that the name of the env variables needs to match the registered name here
+# See authlib documentation for more information
+oauth.register(
+    name="toolhub",
+    access_token_url="https://toolhub-demo.wmcloud.org/o/token/",
```

Comment:
You can delete all of this and leave just `name="toolhub"` here, then plug the URLs into the config file (in the places where we currently have URLs pointing to localhost:5000). I'm the one who confused you with this in the [Phab ticket ](https://phabricator.wikimedia.org/T327856) I'm afraid :/ I was testing a bunch of stuff to make things work, not all were relevant.

What happens is that anything you put inside `oauth.register()` will override whatever is in your config file; conversely, if `oauth.register()` doesn't specify anything, the oauth lib will look inside the config file and get the params from there.

Reply:
That's what I thought would probably happen, but I also thought, "When in doubt let's just put all the things in all the places and then sort it out later."